mapper/orm/src/main/java/org/hibernate/search/mapper/orm/automaticindexing/spi/AutomaticIndexingQueueEventSendingPlan.java

```diff
@@ -45,11 +46,22 @@ public interface AutomaticIndexingQueueEventSendingPlan {
 	 *
 	 * @param <R> The type of entity references in the returned execution report.
 	 * @param entityReferenceFactory A factory for entity references in the returned execution report.
+	 * @param operationSubmitter How to handle request to submit operation when the queue is full.
 	 * @return A {@link CompletableFuture} that will hold an execution report when all the events are sent.
 	 * If sending an event failed, the future will be completed normally,
 	 * but the report will contain an exception.
 	 */
 	<R> CompletableFuture<MultiEntityOperationExecutionReport<R>> sendAndReport(
-			EntityReferenceFactory<R> entityReferenceFactory);
+			EntityReferenceFactory<R> entityReferenceFactory, OperationSubmitter operationSubmitter);
+
+	/**
+	 * @see #sendAndReport(EntityReferenceFactory, OperationSubmitter)
+	 * @deprecated Use {@link #sendAndReport(EntityReferenceFactory, OperationSubmitter)} instead.
+	 */
+	@Deprecated
+	default <R> CompletableFuture<MultiEntityOperationExecutionReport<R>> sendAndReport(
+			EntityReferenceFactory<R> entityReferenceFactory) {
+		return sendAndReport( entityReferenceFactory, OperationSubmitter.BLOCKING );
+	}
```

Comment:
Same comment as elsewhere: this won't preserve backwards compatibility, as this interface is supposed to be *implemented* by service providers.

You'll have to introduce *two* default methods to preserve backwards compatibility (see my other comment about this).